main/autogen/autogen.cc

```diff
@@ -31,6 +31,7 @@ class AutogenWalk {
     vector<core::NameRef> requires;
     vector<DefinitionRef> nesting;
     vector<ast::Send *> ignoring;
+    vector<bool> inBlock;
```

Comment:
Could you give more context behind this change? seems strange to treat includes in blocks and outside blocks differently. Seems like we're papering over a problem, but I might be missing context

That said, you own this code, so it's your choice.

Reply:
The existing implementation won't count references for cases like this:

```
module A
  included do
    include Foo
  end
end
```

as the existing implementation in `autogen.cc` will always ignore sends for `include` and `extend`. The reason for ignoring them is that they're processed explicitly in `preTransformClassDef` by walking the ancestors, so processing them again in sends would count them twice.

This PR fixes that by keeping a stack of booleans that indicates whether or not names are currently being processed in a block. If they are present in a block, then they aren't ignored.